### PR TITLE
John6797/sc 3881/support for metadata in new metric decorator

### PR DESCRIPTION
**Pull request by johnwalz97**
## Internal Notes for Reviewers

Adding new decorators to support task type and tag metadata in functional metrics.

Here is an example:
```python
# Copyright © 2023-2024 ValidMind Inc. All rights reserved.
# See the LICENSE file in the root of this repository for details.
# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial

from sklearn.metrics import accuracy_score

from validmind import tags, tasks


@tasks("classification")
@tags("classification", "sklearn", "accuracy")
def Accuracy(dataset, model):
    """Calculates the accuracy of a model"""
    return accuracy_score(dataset.y, dataset.y_pred(model))


# the above decorator is syntactic sugar for the following:
Accuracy.__tags__ = ["classification"]
Accuracy.__tasks__ = ["classification", "sklearn", "accuracy"]
```

## External Release Notes

Adding new decorators to support task type and tag metadata in functional metrics.

### CODE REVIEW

**Review by cachafla — APPROVED** [review]
👌

**Comment by nrichers** [comment]
@johnwalz97 I think you'd tagged me for review and them removed me. I didn't see anything here that would need my input but as a FYI, I have some edits incoming for these notebooks:

- notebooks/code_samples/custom_tests/external_test_providers.ipynb ([edited version](https://github.com/validmind/developer-framework/blob/nrichers/sc-3795/add-explanations-definition-of-concepts-into/notebooks/code_samples/custom_tests/implement_custom_tests.ipynb))
- notebooks/how_to/run_unit_metrics.ipynb ([edited version](https://github.com/validmind/developer-framework/blob/nrichers/sc-3795/add-explanations-definition-of-concepts-into/notebooks/how_to/run_unit_metrics.ipynb))

Nothing for you to do unless you want to be involved in reviewing these edits when I post the PR.

File: validmind/tests/__init__.py [hunk 1]
```diff
@@ -253,67 +256,71 @@ def list_tests(filter=None, task=None, tags=None, pretty=True, truncate=True):
     return tests
 
 
-def load_test(test_id, reload=False):  # noqa: C901
-    # Extract the test ID extension from the actual test ID when loading
-    # the test class. This enables us to generate multiple results for
-    # the same tests within the document. For instance, consider the
-    # test ID "validmind.data_validation.ClassImbalance:data_id_1,"
```
Comment (AnilSorathiya) [inline validmind/tests/__init__.py:260]: Is there any reasons to remove this comments unless you are handling at different place?  The `result_id` specific comment is useful.

**Review by AnilSorathiya — APPROVED** [review]
LGTM 👏

File: validmind/tests/__init__.py (hunk 1 above)
Comment (johnwalz97) [reply validmind/tests/__init__.py:260]: Nope - that was an oversight, thanks for catching. I added a docstring to the function but I think we should move the test id parsing and complexity to a class eventually.
